scripts/generate-abi.py

```diff
@@ -0,0 +1,77 @@
+import json
+import tempfile
+from pathlib import Path
+
+import click
+from psutil import Popen
+
+_COMPILE_COMMAND = """
+git clone --no-checkout git@github.com:beamer-bridge/beamer.git {beamer_path} &&
+cd {beamer_path} &&
+git checkout {commit_id} &&
+ape compile
+"""
+
+
+def _get_contract_commit_ids(deployment_path: Path) -> dict[str, set[str]]:
+    file_contract_commits: dict[str, set[str]] = {}
```

Comment:
Nitpick: I would just name this `contract_commits`, not sure what "file_" is supposed to mean here.